gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/DiagnosticsManifestPlugin.java

```diff
@@ -95,39 +93,23 @@ public void apply(Project project) {
         });
     }
 
-    private static void configureExternalDependencyTransform(Project project) {
+    private static void configureDiagnosticsTransform(Project project) {
         project.getDependencies().registerTransform(ExtractFileFromJar.class, details -> {
             details.getParameters().getPathToExtract().set(FILE_TO_EXTRACT);
 
             details.getFrom().attribute(artifactType, ArtifactTypeDefinition.JAR_TYPE);
-            details.getTo().attribute(artifactType, MY_ATTRIBUTE);
+            details.getTo().attribute(artifactType, DIAGNOSTICS);
 
+            // It's not strictly necessary to define a mapping for the 'org.gradle.libraryelements' attribute,
+            // but it looks a bit weird for something to still be marked as a 'jar' when it's clearly not.
             details.getFrom()
                     .attribute(
                             LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE,
                             project.getObjects().named(LibraryElements.class, LibraryElements.JAR));
             details.getTo()
                     .attribute(
                             LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE,
-                            project.getObjects().named(LibraryElements.class, MY_ATTRIBUTE));
-        });
-    }
-
-    private static void configureProjectDependencyTransform(Project project) {
```

Comment:
My initial approach did not work because we were registering multiple artifact transforms. Both the resources directory and the classes directory transform would produce a dependency with variant

```
artifactType=diagnostics,org.gradle.libraryelements=diagnostics
```

During resolution, Gradle must select one. There are some docs that vaguely describe how this works. In my testing the resources directory was chosen fairly consistently over the classes directory.
 
https://docs.gradle.org/7.6/userguide/artifact_transforms.html#artifact_transform_selection_and_execution

The only way to make this work would be to define separate attribute values, so the variants are different. For example, a dependency would produce both:
```java
// These are from the resources directory
artifactType=diagnostics,org.gradle.libraryelements=resource-diagnostics

// These are from the class directory
artifactType=diagnostics,org.gradle.libraryelements=class-diagnostics
```

_Interesting, because a local dependency also produces a JAR, I believe the JAR transform also conflicts with the resources directory transform._

This solution would also require us to define a separate artifact view for each variant and pass those all into the classpath for the `mergeDiagnosticsTask`.

Looking back at code comments from https://github.com/palantir/sls-packaging/pull/1035, it looks like this was done purely as an optimization to avoid needing to create a JAR to extract the diagnostic manifest.

https://github.com/palantir/sls-packaging/blob/614f8a265085d3fcdf32e70c81b7c56ef4da9ac0/gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/service/DiagnosticsManifestPlugin.java#L107-L109

It feels reasonable to just eliminate this optimization, with the following justification:
- It is undesirable to have conflicting artifact transforms
- Removing it makes the plugin logic simpler
- Packaging JARs is relatively fast
- Most of the builds that run `createManifest` are probably going to end up packaging the JAR anyways

Reply:
I agree, this is entirely reasonable. We have several other components which tie into distribution/manifest creation (e.g. minimum dependency calculation based on referenced rpc methods) which also require compilation.

Reply:
I'd need to check, but my one fear with making createManifest depend on compilation is that dev-env might delay the `resolveSlsConfigurations` task, which will slow down builds by minutes as we can't pull containers early (requires `resolveSlsConfigurations` to be run). It also means `resolveSlsConfigurations` and it successors (`gofigure` etc) cannot be run in parallel with compilation and must come after it.

Reply:
> We have several other components which tie into distribution/manifest creation (e.g. minimum dependency calculation based on referenced rpc methods) which also require compilation.

I'm pretty all the scary code involving artifacts transforms inside this repo was to avoid that.

Reply:
I don't see how you can avoid compilation if we want to support the annotation processor.

I don't completely understand how we have this guarantee today. It's not obvious to me what would make Gradle reliably choose the resources directory artifact transform over the JAR artifact transform.

Just looking at this code and the [Gradle docs](https://docs.gradle.org/7.6/userguide/artifact_transforms.html#artifact_transform_selection_and_execution), I would have thought that we'd fall into the "selection fails and an error is reported" case. But that's obviously not the case.

> Of all the found transform chains, Gradle tries to select the best one:
> - If there is only one transform chain, it is selected.
> - If there are two transform chains, and one is a suffix of the other one, it is selected.
> - If there is a shortest transform chain, then it is selected.
> - In all other cases, the selection fails and an error is reported.

Reply:
Talking @carterkozak, we're rolling out the conjure per-endpoint minimum version plugin atm, which will force compilation to occur anyway. This will slow down builds, but I think I can make another task in sls-packaging to spit out the pdeps to file, then have dev-env read that rather than `createManifest`. This way we can make builds fast again. So I think this is fine.